src/logic/safe/store/actions/utils.ts

```diff
@@ -120,3 +131,16 @@ export const getNonce = async (safeAddress: string, safeVersion: string): Promis
   }
   return nextNonce
 }
+
+export const navigateToTx = (safeAddress: string, txDetails: TransactionDetails): void => {
```

Comment:
Nice idea to move this function, but why to the store/actions/utils? It has more to do with routing.

Reply:
I wasn't sure where to move it so I moved it to the closest utils file. I agree it would be better to have it somewhere with routing maybe in `src/routes/routes.ts`?

Reply:
I would not concern yourself about moving this as it is purely transaction creation focused and not related to testing.

When we refactor this properly, we should also consider supporting deeplinking via `txId`s as it is a) part of the deeplinking v2 and b) will heavily simplify this function.